### ISSUE TITLE
[TableOfContents]: incorrect role usage

### ISSUE THREAD

**Issue by samblikArendus** [TEDI-Design-System/react#539]
## Problem
TableOfContents uses role="tree" which is not meant for this type of element (go [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tree_role) for more info). Currently the element is failing WCAG.
## Solution
Remove  incorrect role usage and add aria-label for instance to mark it as table of contents.
